### PR TITLE
[ARTS-352] Add Discovery Server service

### PR DESCRIPTION
**Pull request by anatbal**
## Description
Adding discovery server service, with no wiring except of adding the service to CI

### Changes
- [ARTS-352] - Adding discovery server 

### Checklist:

- [x] Branch name follows convention (feature/arts-###-short-name OR fix/arts-###-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

[ARTS-352]: https://ndph-arts.atlassian.net/browse/ARTS-352

### CODE REVIEW

**Review by eladiw — DISMISSED** [review]
LGTM

File: discovery-server/pom.xml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.4.2</version>
```
Comment (tamirkamara) [inline discovery-server/pom.xml:7]: This isn't the version we use on the other modules. Can you refer to the config server and check both the spring-boot version and the cloud version?

File: discovery-server/pom.xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+    <repositories>
+        <repository>
+            <id>spring-milestones</id>
+            <name>Spring Milestones</name>
+            <url>https://repo.spring.io/milestone</url>
+        </repository>
+    </repositories>
```
Comment (tamirkamara) [inline discovery-server/pom.xml:47]: Do we need this?

File: pom.xml [hunk 3]
```diff
@@ -33,6 +33,7 @@
         <module>sample-service</module>
         <module>practitioner-service</module>
         <module>config-server</module>
+        <module>discovery-server</module>
```
Comment (tamirkamara) [inline pom.xml:36]: can you move both of these below the "core" services?

File: discovery-server/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+# cache most dependencies as a layer to be used in other builds while POMs don't change
+RUN mvn dependency:go-offline --batch-mode -q --projects ${SVC}
+
+# copy current service source
+COPY ./${SVC}/src ${SVC}/src
+
+RUN mvn package --projects ${SVC} # -Dmaven.test.failure.ignore=true
+
+FROM scratch AS export
+ARG SVC
```
Comment (tamirkamara) [inline discovery-server/Dockerfile:26]: Since you don't do anything here... you can delete these 2 lines.

File: discovery-server/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+COPY ./init-service/pom.xml ./init-service/pom.xml
+COPY ./site-service/pom.xml ./site-service/pom.xml
+
+# cache most dependencies as a layer to be used in other builds while POMs don't change
+RUN mvn dependency:go-offline --batch-mode -q --projects ${SVC}
+
+# copy current service source
+COPY ./${SVC}/src ${SVC}/src
+
+RUN mvn package --projects ${SVC} # -Dmaven.test.failure.ignore=true
```
Comment (tamirkamara) [inline discovery-server/Dockerfile:23]:
```suggestion
RUN mvn package --projects ${SVC}
```

**Review by tamirkamara — CHANGES_REQUESTED** [review]
Overall looks good. The main comment is about the spring boot/cloud versions.

File: discovery-server/pom.xml (hunk 1 above)
Comment (anatbal) [reply discovery-server/pom.xml]: Yes you are right, I though I fixed it but apparently it was in the second PR. Changed to version 2.3.6.RELEASE. The cloud version is okay, Hoxton.SR9

File: discovery-server/pom.xml (hunk 2 above)
Comment (anatbal) [reply discovery-server/pom.xml]: No, fixed

File: pom.xml (hunk 3 above)
Comment (anatbal) [reply pom.xml]: Done

File: discovery-server/Dockerfile (hunk 4 above)
Comment (anatbal) [reply discovery-server/Dockerfile]: Done

File: discovery-server/Dockerfile (hunk 5 above)
Comment (anatbal) [reply discovery-server/Dockerfile]: Done

**Review by tamirkamara — APPROVED** [review]

**Review by darrengsmithson — APPROVED** [review]

**Pull request merged**